docs/debug/2_config_file_structure.ipynb

```diff
@@ -0,0 +1,241 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Config File Structure\n",
+    "\n",
+    "To enable debug features, create a `config.yaml` file to specify the desired behavior, such as determining which GEMMs (General Matrix Multiply operations) should run in higher precision rather than FP8 and defining which statistics to log. Below, we outline how to structure the `config.yaml` file.\n",
+    "\n",
+    "## General Format\n",
+    "\n",
+    "A config file can have one or more sections, each containing settings for specific layers and features:\n",
+    "\n",
+    "```yaml\n",
+    "section_name_1:\n",
+    "  enabled: ...\n",
+    "  layers:\n",
+    "    # Specify layers here...\n",
+    "  transformer_engine:\n",
+    "    Feature1Name:\n",
+    "      enabled: ...\n",
+    "      # Feature details...\n",
+    "    Feature2Name:\n",
+    "      enabled: ...\n",
+    "      # Feature details...\n",
+    "\n",
+    "section_name_2:\n",
+    "  enabled: ...\n",
+    "  layers:\n",
+    "    # Specify layers here...\n",
+    "  transformer_engine:\n",
+    "    Feature1Name: # If feature has no namespace, then it is in the default namespace.\n",
+    "      enabled: ...\n",
+    "      # Feature details...\n",
+    "\n",
+    "section_name_3:\n",
+    "  enabled: ...\n",
+    "  layers:\n",
+    "    # Specify layers here...\n",
+    "  transformer_engine:\n",
+    "    Feature1Name:\n",
+    "      enabled: ...\n",
+    "      # Feature details...\n",
+    "    Feature2Name:\n",
+    "      enabled: ...\n",
+    "      # Feature details...\n",
+    "```\n",
+    "\n",
+    "Each section can have any name and must contain:\n",
+    "\n",
+    "1. An `enabled` field that specifies whether the features in that section will be active.\n",
+    "2. A `layers` field specifying which layers the section applies to. Each layer can belong to only one section.\n",
+    "3. Additional fields describing features for those layers.\n",
+    "\n",
+    "## Layer Specification\n",
+    "\n",
+    "Debug layers can be identified by a `name` parameter:\n",
+    "\n",
+    "```python\n",
+    "linear = transformer_engine.debug.pytorch.Linear(in_features, out_features, name=\"linear1\")\n",
+    "```\n",
+    "\n",
+    "This name is used in the config file to identify the layer. To specify the `layers` field, you can use one of the following methods:\n",
+    "\n",
+    "1. **`layer_name_regex_pattern`**: Use a regex to match layer names.\n",
```

Comment:
```suggestion
    "1. **`layer_name_regex_pattern`**: Use a regular expression to match layer names.\n",
```

Reply:
It would also probably be good to say what the syntax of the regular expression should be.